addons/mail/static/src/models/message/message.js

```diff
@@ -583,15 +561,31 @@ function factory(dependencies) {
         subject: attr(),
         subtype_description: attr(),
         subtype_id: attr(),
-        threadCaches: many2many('mail.thread_cache', {
+        /**
+         * All threads that this message is linked to.
+         */
+        threads: many2many('mail.thread', {
+            compute: '_computeThreads',
+            dependencies: [
+                'isHistory',
+                'isModeratedByCurrentPartner',
+                'isNeedaction',
+                'isStarred',
+                'originThread',
+                '_threads',
+            ],
             inverse: 'messages',
         }),
-        threadCachesThread: many2many('mail.thread', {
-            related: 'threadCaches.thread',
-        }),
         tracking_value_ids: attr({
             default: [],
         }),
+        /**
+         * All threads that this message is linked to, without considering the
+         * origin thread.
+         */
+        _threads: many2many('mail.thread', {
```

Comment:
This field contains `mail.channel` threads from server, and also mailboxes.
I added prefix `_` just to avoid name clashing with `threads`, the latter being the one overly used for read. `_threads` is mainly used for "writes".

I think I could improve that by splitting concern of this field from `convertData` into a new field `serverChannels`. As for the links to other threads... I'm not entirely sure, but maybe we should have a `nonOriginThreads` semi-computed field:

- computed method to have `serverChannels` and linked mailboxes + never has `originThread`
- write on this field to link non-origin threads

We could also introduce a feature to customize `.update()` on a field, so that update on `threads` could auto-update either `nonOriginThreads` or `originThread` based on presence of provided threads to link.

All of this could be very nice, as each field is meaningful with this design.

Reply:
In the end, I just split `_threads` into `serverChannels` and `nonOriginThreads`. I didn't make customizable `.update()` for threads, because in attempting to do it, this made things overly complicated to manage, which is a sign this was a bad idea.

I've put a mention that `nonOriginThreads` and `threads` are read-only fields, would certainly consider adding a read-only option on fields in the future.

Reply:
Ok this is good